torch/csrc/jit/passes/python_print.cpp

```diff
@@ -1010,15 +1010,30 @@ struct PythonPrintPass {
              << node->i(attr::end) << "]";
       } break;
       case prim::ListConstruct: {
-        // when the list is empty and is not a list of tensors,
-        // we need to annotate it, otherwise it won't be possible
-        // to infer the type on import
-        if (node->inputs().size() == 0 &&
-            !node->output()->type()->isSubtypeOf(TensorType::get())) {
-          stmt << "annotate(" << node->output()->type()->python_str()
-               << ", [])";
-        } else {
+        if (ListTypePtr list_type = node->output()->type()->cast<ListType>()) {
+          TypePtr elem_type = list_type->getElementType();
+          if (!elem_type->isSubtypeOf(TensorType::get())) {
+            // when the list is empty and is not a list of tensors,
+            // we need to annotate it, otherwise it won't be possible
+            // to infer the type on import
+            if (node->inputs().size() == 0) {
+              stmt << "annotate(" << node->output()->type()->python_str()
+                   << ", [])";
+              break;
```

Comment:
These non-local breaks are hard to follow. Can you avoid them? It is easy to miss the fact that control ends here.